src/pgroonga.c

```diff
@@ -6377,10 +6377,23 @@ PGrnSearchBuildCondition(Relation index, ScanKey key, PGrnSearchData *data)
 			keywordSize =
 				grn_vector_get_element(ctx, keywords, i, &keyword, NULL, NULL);
 			GRN_TEXT_SET(ctx, &keywordBuffer, keyword, keywordSize);
+			if (keywordSize == 0 && operator== GRN_OP_REGEXP)
+			{
+				continue;
+			}
 			PGrnSearchBuildConditionBinaryOperation(
 				data, targetColumn, &keywordBuffer, operator);
-			if (i > 0)
-				PGrnExprAppendOp(data->expression, GRN_OP_OR, 2, tag, NULL);
+			if (operator== GRN_OP_REGEXP)
+			{
+				nTargetKeywords++;
+				if (i > 0 && nTargetKeywords > 1)
+					PGrnExprAppendOp(data->expression, GRN_OP_OR, 2, tag, NULL);
+			}
+			else
+			{
+				if (i > 0)
+					PGrnExprAppendOp(data->expression, GRN_OP_OR, 2, tag, NULL);
+			}
```

Comment:
```suggestion
			if (nTargetKeywords > 0)
				PGrnExprAppendOp(data->expression, GRN_OP_OR, 2, tag, NULL);
			nTargetKeywords++;		
```

`operator== GRN_OP_REGEXP` で場合分けしなくても、こういう感じで動かないですか？

Reply:
あぁ。たしかに。↑で動きます。
↑のように直してテストしてみます。

Reply:
手元のテストはすべて通ったので↑のように修正しました。